### PR TITLE
Use unique reference in redux ID

### PR DESCRIPTION
**Pull request by spadgett**
/cc @alecmerdler @christianvogt 

@invincibleJai are you able to help me test this fix?

https://jira.coreos.com/browse/CONSOLE-1550

### CODE REVIEW

**Comment by alecmerdler** [comment]
Of course this would be the cause. We've needed this for a while. Good catch.

**Comment by invincibleJai** [comment]
@spadgett sure i'll do it, i was about to file a separate PR as https://github.com/openshift/console/pull/1800 has lot of other changes 

and adding `kind` in id would make it unique.

**Comment by invincibleJai** [comment]
Looks good to me, verified !!

Screenshot 2019-06-26 at 8 00 13 AM

**Comment by spadgett** [comment]
@alecmerdler It looks like there are cases where `k8sKind` is `undefined` in `makeReduxID` (at least the Installed Operators page). I'm investigating.

**Comment by christianvogt** [comment]
@spadgett if it were undefined, then the old code would have also been failing.

**Comment by spadgett** [comment]
> @spadgett if it were undefined, then the old code would have also been failing.

It is sometimes `undefined`, but digging deeper that doesn't seem to be the problem. Looks like it's the check here:

https://github.com/spadgett/console/blob/cd47343e287bd07b5ab721da51e4afcb07295fdc/frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx#L127

**Comment by spadgett** [comment]
@alecmerdler I don't believe the check is needed anymore after #776. We should be able to remove that `stateToProps` entirely. I see no flicker.

File: frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx [hunk 1]
```diff
@@ -123,16 +123,7 @@ export const ClusterServiceVersionList: React.SFC<ClusterServiceVersionListProps
...
-  loading: FLAGS.get(featureFlags.OPENSHIFT) === undefined || !k8s.getIn([FLAGS.get(featureFlags.OPENSHIFT) ? 'projects' : 'namespaces', 'loaded']),
-});
-
-export const ClusterServiceVersionsPage = connect(stateToProps)((props: ClusterServiceVersionsPageProps) => {
-  // Wait for OpenShift feature detection to prevent flash of "disabled" UI
-  if (props.loading) {
-    return <LoadingBox />;
-  }
-
+export const ClusterServiceVersionsPage = (props: ClusterServiceVersionsPageProps) => {
```
Comment (alecmerdler) [inline frontend/public/components/operator-lifecycle-manager/clusterserviceversion.tsx]: Going to need to remove `.WrappedComponent` from [the unit test as well](https://github.com/openshift/console/blob/master/frontend/__tests__/components/operator-lifecycle-manager/clusterserviceversion.spec.tsx#L127).

**Comment by spadgett** [comment]
@alecmerdler thanks, updated

**Comment by spadgett** [comment]
CI is green now

**Comment by spadgett** [comment]
Rebased

**Comment by spadgett** [comment]
/hold
to prevent retest. vertical pod autoscaler broke e2e-aws earlier today

**Pull request merged**